nwb_conversion_tools/utils/neonwbephyswriter.py

```diff
@@ -0,0 +1,110 @@
+from abc import abstractmethod
+from distutils.version import StrictVersion
+
+import pynwb
+
+from .basenwbephyswriter import BaseNwbEphysWriter
+
+try:
+    import neo
+
+    if StrictVersion(neo.__version__) >= StrictVersion("0.10"):
+        HAVE_NEO = True
+    else:
+        HAVE_NEO = False
+except ImportError:
+    HAVE_NEO = False
+
+
+class NEONwbEphysWriter(BaseNwbEphysWriter):
+    """
+    Class to write a neo.RawIO or neo.IO versio>=0.10 to NWB
+
+    Parameters
+    ----------
+    object_to_write: neo.RawIO or neo.IO
+    stub: bool
+        whether to write a subset of recording extractor traces array as electrical series in nwbfile
+    stub_channels: list
+        channels to include when writing as stub
+    """
+
+    def __init__(self, object_to_write, stub=False, stub_channels=None):
+        assert HAVE_NEO
+        self.recording, self.sorting, self.event = None, None, None
+        BaseNwbEphysWriter.__init__(self, object_to_write, stub=stub, stub_channels=stub_channels)
+
+    @staticmethod
+    def supported_types():
+        assert HAVE_NEO
+        return (neo.rawio.baserawio.BaseRawIO, neo.io.baseio.BaseIO)
+
+    def add_to_nwb(self, nwbfile: pynwb.NWBFile, metadata=None, **kwargs):
+        # check what's in the neo object: analogsignals, spike trains, events and
+        # write recording, sorting, events accordingly
+        raise NotImplementedError
+
+    def add_recording(self, segment_index=0):
+        raise NotImplementedError
+
+    def add_sorting(self):
+        raise NotImplementedError
+
+    def add_waveforms(self):
+        raise NotImplementedError
+
+    def add_epochs(self):
+        raise NotImplementedError
+
+    @abstractmethod
```

Comment:
why is this abstract?

Reply:
yes correct `make_recording_stub()` and `make_sorting_stub()` should be moved to the base class: `BaseNwbEphysWriter`. NeoEphysWriter will them implement this.

Reply:
https://github.com/catalystneuro/nwb-conversion-tools/pull/230/commits/d7cb6236c547ac89bee3be6d759121b314c464cb